### PR TITLE
[Docs] Add BEV-based detection pipeline in NuScenes Dataset tutorial

### PR DESCRIPTION
**Pull request by 1uciusy**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

I was learning to use `mmdetction3d`. When I was trying to use sample codes in the tutorial of NuScenes Dataset, the samples were not accessible from the dataset by index. I looked into the tracebacks and found the `train_pipeline` has not been quite updated. To avoid similar problem other beginners may encounter, I want to update the tutorial

## Modification

Please briefly describe what modification is made in this PR.

Update the `train_pipeline` definition.

- `LoadImageFromFileMono3D` should be replaced by `LoadMultiViewImageFromFiles`. It's been discussed in many issues.
- `LoadAnnotations3D` is set to default, otherwise there would be tracebacks, for example, the following traceback will happen if `with_bbox=True`:
```
  File "/mnt/filesystem3/yangyi/anaconda3/envs/re_bevformer/lib/python3.9/site-packages/mmdet/datasets/transforms/loading.py", line 396, in transform
    self._load_bboxes(results)
  File "/home/yangyi/wd/git_repo/mmdetection3d/mmdet3d/datasets/transforms/loading.py", line 1033, in _load_bboxes
    results['gt_bboxes'] = results['ann_info']['gt_bboxes']
KeyError: 'gt_bboxes'
```
- `mmdet.Resize` is removed. The `mmdet.Resize` directly change `results['img']` as a `numpy.ndarray`, while in 3D detection, `result['img']` is a list of `numpy.ndarray`, the following traceback would happen:
```
  File "/mnt/filesystem3/yangyi/anaconda3/envs/re_bevformer/lib/python3.9/site-packages/mmdet/datasets/transforms/transforms.py", line 154, in transform
    self._resize_img(results)
  File "/mnt/filesystem3/yangyi/anaconda3/envs/re_bevformer/lib/python3.9/site-packages/mmcv/transforms/processing.py", line 172, in _resize_img
    img, scale_factor = mmcv.imrescale(
  File "/mnt/filesystem3/yangyi/anaconda3/envs/re_bevformer/lib/python3.9/site-packages/mmcv/image/geometric.py", line 279, in imrescale
    h, w = img.shape[:2]
AttributeError: 'list' object has no attribute 'shape'
```
- `RandomFlip3D` is removed. The `RandomFlip3D` stacks all six views of images and change `result["img"]` to a `numpy.ndarray` with shape of (6, 900, 1600, 3), causing problem in following transforms. The traceback would be similar to the following:
```
  File "/mnt/filesystem3/yangyi/anaconda3/envs/re_bevformer/lib/python3.9/site-packages/mmcv/transforms/base.py", line 12, in __call__
    return self.transform(results)
  File "/home/yangyi/wd/git_repo/mmdetection3d/mmdet3d/datasets/transforms/formating.py", line 119, in transform
    return self.pack_single_results(results)
  File "/home/yangyi/wd/git_repo/mmdetection3d/mmdet3d/datasets/transforms/formating.py", line 169, in pack_single_results
    np.ascontiguousarray(img.transpose(2, 0, 1)))
ValueError: axes don't match array
```

With the left transformation `LoadMultiViewImageFromFiles`, `LoadAnnotations3D` and `Pack3DDetInputs`, the pipeline can still finish the basic job of reading multi-view images, loading ground truth and packing.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by sunjiahao1999** [comment]
It's the common pipeline for mono 3d model like FCOS3D, you should keep thie pipeline and add a new common BEV pipeline as your PR.Besides, add a note to the describe their difference. Thank you for your contribution to MMDetection3D community.

**Comment by 1uciusy** [comment]
@sunjiahao1999 OK, I will add an alternative BEV pipeline in the tutorial. 

I also noticed there is some strange behaviors in the dataset `NuScenesDataset` when I try to load all six views. When I run `self.get_data_info(idx)`, a method called in `self.__getitem__(idx)`, image path should be like `.data/nuscenes/samples/CAM_FRONT/XXX.jpg`, instead, I got `.data/nuscenes/XXX.jpg`, thus I cannot load the images naturally through `self.__getitem__(idx)`.  issues. I volunteer to give my solution. Should I commit in this PR or start a new one?

File: docs/en/advanced_guides/datasets/nuscenes.md
Comment (1uciusy) [reply docs/en/advanced_guides/datasets/nuscenes.md]: Hi, my bad. I've restored the monocular part but changed a bit where Chinese version may not be correct.

Also, I add a basic BEV training pipeline as an alternative and description about the difference between BEV and monocular. Necessary modification to dataset initialization to utilize BEV pipeline is also added.

**Comment by ZCMax** [comment]
Thanks for your PR, if we want to introduce the BEV series perception paradigm in the tutorial, it's better to use two subsections to introduce the vision-based pipeline:
1. monocular-based  
    In the NuScenes dataset, for multi-view images, this paradigm usually involves detecting and outputting 3D object detection results separately for each image, and then obtaining the final detection results through post-processing (such as NMS). Essentially, it directly extends monocular 3D detection to multi-view settings.
3. BEV-based
   BEV, Bird's-Eye-View, is another popular 3D detection paradigm. It directly takes multi-view images to perform 3D detection, for nuScenes, they are `CAM_FRONT`, `CAM_FRONT_LEFT`, `CAM_FRONT_RIGHT`, `CAM_BACK`, `CAM_BACK_LEFT` and `CAM_BACK_RIGHT`. A basic training pipeline of bev-based 3D detection on nuScenes is as below.

File: docs/en/advanced_guides/datasets/nuscenes.md
Comment (Xiangxu-0103) [inline docs/en/advanced_guides/datasets/nuscenes.md:156]: better to use `####` to add sub-section.

File: docs/en/advanced_guides/datasets/nuscenes.md
Comment (Xiangxu-0103) [inline docs/en/advanced_guides/datasets/nuscenes.md:157]: Add a blank above.

File: docs/en/advanced_guides/datasets/nuscenes.md
Comment (Xiangxu-0103) [inline docs/en/advanced_guides/datasets/nuscenes.md:189]: The same.

File: docs/en/advanced_guides/datasets/nuscenes.md
Comment (Xiangxu-0103) [inline docs/en/advanced_guides/datasets/nuscenes.md:196]: Better to add some common `aug` usage for BEV-based detection, such as `Resize3D`

File: docs/en/advanced_guides/datasets/nuscenes.md
Comment (Xiangxu-0103) [inline docs/en/advanced_guides/datasets/nuscenes.md:214]: We suggest to use config style, such as `train_dataloader=dict(dataset=dict(...))`

Comment (Xiangxu-0103) [reply docs/en/advanced_guides/datasets/nuscenes.md:214]: You may refer to this [config](https://github.com/open-mmlab/mmdetection3d/blob/main/projects/DETR3D/configs/detr3d_r101_gridmask.py) for reference. Also, we will consider to add a basic dataset config for BEV-based detection on nus.

**Comment by 1uciusy** [comment]
@Xiangxu-0103 Hi, this PR requires approval from a maintainer to be merged, could you review my code?

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**
